src/plugins/discover/server/plugin.ts

```diff
@@ -36,6 +36,15 @@ import { searchSavedObjectType } from './saved_objects';
 export class DiscoverServerPlugin implements Plugin<object, object> {
   public setup(core: CoreSetup) {
     core.capabilities.registerProvider(capabilitiesProvider);
+    core.capabilities.registerSwitcher(async (request, capabilites) => {
```

Comment:
As there is no consumer under the line of await, I'd recommend removing the async / await in this function, not a big concern.

Reply:
@SuZhou-Joe `hideForReadonly` is async and it's needed to be due to implementation from security plugin site

Reply:
Actually because there is no consumer following the await, so it's the same to make it from async / await to a simple () => { return core.security.readonly } because both will return a promise to the function caller, but still, not a big concern.

Reply:
I would rather keep it as it is